iothub/device/src/Transport/AmqpIoT/AmqpIoTReceivingLink.cs

```diff
@@ -258,88 +258,104 @@ private void DisposeDelivery(AmqpMessage amqpMessage, bool settled, Accepted acc
+                            string patch = reader.ReadToEnd();
+                            twinProperties = JsonConvert.DeserializeObject<TwinCollection>(patch);
+                        }
+                    }
+                    else if (correlationId.StartsWith(AmqpTwinMessageType.Get.ToString(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        // This a response of a GET TWIN so return (set) the full twin
                         using (StreamReader reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8))
                         {
                             string body = reader.ReadToEnd();
                             var properties = JsonConvert.DeserializeObject<TwinProperties>(body);
                             twin = new Twin(properties);
                         }
                     }
-                    else
+                    else if (correlationId.StartsWith(AmqpTwinMessageType.Patch.ToString(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        // This can be used to coorelate success response with Updating reported properties
+                    }
+                    else if (correlationId.StartsWith(AmqpTwinMessageType.Put.ToString(), StringComparison.OrdinalIgnoreCase))
                     {
-                        // This is a desired property ack from the service
+                        // This is an acknowledgement received from service for subscribing to desired property updates
                         twin = new Twin();
                     }
-                }
-                else
-                {
-                    // No correlationId, this is a PATCH sent by the sevice so return (set) the TwinCollection
-
-                    using (StreamReader reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8))
+                    else
                     {
-                        string patch = reader.ReadToEnd();
-                        twinProperties = JsonConvert.DeserializeObject<TwinCollection>(patch);
+                        // This shouldn't happen
+                        // Do nothing
                     }
+                    _onDesiredPropertyReceived.Invoke(twin, correlationId, twinProperties);
                 }
-                _onDesiredPropertyReceived.Invoke(twin, correlationId, twinProperties);
             }
             finally
             {
                 if (Logging.IsEnabled)
                 {
-                    Logging.Exit(this, amqpMessage, $"{nameof(OnDesiredPropertyReceived)}");
+                    Logging.Exit(this, amqpMessage, $"{nameof(OnTwinChangesReceived)}");
                 }
             }
         }
 
         #endregion Twin handling
 
-        internal static bool VerifyResponseMessage(AmqpMessage response)
+        internal static int getStatus(AmqpMessage response)
         {
-            bool retVal = true;
+            int retStatus = -1;
```

Comment:
retStatus is only being used once to return. Maybe not needed and just do a return -1;